system/jlib/jptree.cpp

```diff
@@ -1794,7 +1794,13 @@ IPropertyTree *PTree::addPropTree(const char *xpath, IPropertyTree *val)
                 IPropertyTree *_val = ownPTree(val);
                 dbgassertex(QUERYINTERFACE(_val, PTree));
                 PTree *__val = static_cast<PTree *>(_val);
+
+                /* NB: potentially param xpath is a reference to the existing name.
+                 * So fetch new name ptr after set.
+                 */
                 __val->setName(xpath);
```

Comment:
I could test __val->queryName() ptr vs xpath.

But I think it is safe like this, in theory it could not equal, but be a ptr to an offset into the previous name (however unlikely).